src/info/manifest.c

```diff
@@ -1552,6 +2322,55 @@ manifestFileTotal(const Manifest *this)
     FUNCTION_TEST_RETURN(lstSize(this->fileList));
 }
 
+void
+manifestFileUpdate(
+    Manifest *this, const String *name, uint64_t size, uint64_t sizeRepo, const char *checksumSha1, const String *reference,
+    bool checksumPage, bool checksumPageError, const VariantList *checksumPageErrorList)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(MANIFEST, this);
+        FUNCTION_TEST_PARAM(STRING, name);
+        FUNCTION_TEST_PARAM(UINT64, size);
+        FUNCTION_TEST_PARAM(UINT64, sizeRepo);
+        FUNCTION_TEST_PARAM(STRINGZ, checksumSha1);
+        FUNCTION_TEST_PARAM(STRING, reference);
+        FUNCTION_TEST_PARAM(BOOL, checksumPage);
+        FUNCTION_TEST_PARAM(BOOL, checksumPageError);
+        FUNCTION_TEST_PARAM(VARIANT_LIST, checksumPageErrorList);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+    ASSERT(name != NULL);
+    ASSERT(
+        (!checksumPage && !checksumPageError && checksumPageErrorList == NULL) ||
+        (checksumPage && !checksumPageError && checksumPageErrorList == NULL) || (checksumPage && checksumPageError));
+
+    ManifestFile *file = (ManifestFile *)manifestFileFind(this, name);
+
+    MEM_CONTEXT_BEGIN(lstMemContext(this->fileList))
+    {
+        // Update reference if set
+        if (reference != NULL)
+            file->reference = strLstAddIfMissing(this->referenceList, reference);
+
+        // Update checksum if set
+        if (checksumSha1 != NULL)
+            memcpy(file->checksumSha1, checksumSha1, HASH_TYPE_SHA1_SIZE_HEX + 1);
+
+        // Update repo size
+        file->size = size;
+        file->sizeRepo = sizeRepo;
+
+        // Update checksum page info
+        file->checksumPage = checksumPage;
```

Comment:
checksum-page may not even exist which is not the same as setting it to false, so how/where is that being handled?  
Example from same manifest:
```
pg_data/PG_VERSION={"checksum":"184473f470864e067ee3a22e64b47b0a1c356f29","master":true,"reference":"[BACKUP-FULL-2]","size":[SIZE],"timestamp":[TIMESTAMP-1]}
pg_data/base/1/12000={"checksum":"22c98d248ff548311eda88559e4a8405ed77c003","checksum-page":true,"reference":"[BACKUP-FULL-2]","size":[SIZE],"timestamp":[TIMESTAMP-1]}
pg_data/base/16384/17000={"checksum":"e0101dd8ffb910c9c202ca35b5f828bcb9697bed","checksum-page":false,"checksum-page-error":[1],"reference":"[BACKUP-FULL-2]","size":[SIZE],"timestamp":[TIMESTAMP-1]}
pg_tblspc/1/[TS_PATH-1]/16384/tablespace1.txt={"checksum":"d85de07d6421d90aa9191c11c889bfde43680f0f","checksum-page":false,"size":[SIZE],"timestamp":[TIMESTAMP-1]}

```

Reply:
Search for `kvPut(fileKv, MANIFEST_KEY_CHECKSUM_PAGE_VAR, VARBOOL(!file->checksumPageError));` to find the save logic.

Basically, if it is not in the manifest checksumPage is false, if in manifest then true means checksum errors, and false means no checksum errors.  checksumPageErrorList is present if there are page errors (rather than a validation error).  Yeah, it's complicated, but that's Perl for ya.